app/models/repository/importing.rb

```diff
@@ -153,10 +153,13 @@ def clean_and_initialize_record
   end
 
   def detect_source_type
-    if GitRepository.is_git_repository?(source_address)
-      self.source_type = 'git'
-    elsif GitRepository.is_svn_repository?(source_address)
-      self.source_type = 'svn'
+    if source_address?
+      self.source_type =
+        if GitRepository.is_git_repository?(source_address)
+          'git'
+        elsif GitRepository.is_svn_repository?(source_address)
+          'svn'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.